### PR TITLE
[feature](meta) queries as table valued function (#)

### LINKED ISSUE #25051: [Enhancement] Running Queries as table valued function
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Running queries and history queries are viewed via Rest interface /api/query_detail
However, this does not allows many filters or grouping on the running queries or history queries.

### Solution

Enable queries view through Table Valued Functions similar to frontends()/backends()/workload_groups() etc.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Nitin-Kashyap**
## Proposed changes
1. Add queries view as table function.
2. Proxy result to other FEs and return merged results back to BE.

Issue Number: close #25051

image

### CODE REVIEW

File: gensrc/thrift/PlanNodes.thrift [hunk 1]
```diff
@@ -465,11 +465,17 @@ struct TFrontendsMetadataParams {
   1: optional string cluster_name
 }
 
+struct TQueriesMetadataParams {
+  1: optional string cluster_name
+  2: optional bool   relayToOtherFE
```
Comment (yiguolei) [inline gensrc/thrift/PlanNodes.thrift:470]: according to the code ruls, please rename relayToOtherFE to relay_to_other_fe

Comment (Nitin-Kashyap) [reply gensrc/thrift/PlanNodes.thrift]: fixed

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Review by hf200012 — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**